### DISCUSSION TITLE
Run code-server in Windows

### DISCUSSION THREAD

**Discussion by lg-nineoneninethreenine** [coder/code-server#3035] [Q&A] [answered]
Can I run this in Windows? Like, I don't have a Mac or a Linux distro and I don't wanna pressurise my Android device. Can I run this in Windows's CLI? Because, I want to have VS Code on the cloud, so I can edit it in any of my devices, that's why.

**Comment by bpmct** [comment, marked as answer]
@fearthecowboy made a pretty awesome WSL tutorial for code-server which makes it possible to run in Windows: https://fearthecowboy.com/tutorial/2020/09/17/Code-Serrver-On-WSL.html

You can also run code-server on a remote linux machine/container and then access it from any web browser. We have some scripts and guides on https://github.com/cdr/deploy-code-server

**Reply by lg-nineoneninethreenine** [reply]
This is fine, but can I run it on Windows 7?

**Reply by nooobcoder** [reply]
@bpmct said it correctly
Deep inside you need to be involved in some sort of linux VMs even if you use docker
**So honestly not possible on windows 7 standalone**
Possible if you Install docker and run in it
But still docker would require virtialization enabled and hog up your resources on a windows machine

Better to have a linux machine on the cloud or on premise if you can setup
My suggestion: you can use Google Cloud, you get 300$ credits on signing up and VMs come very cheap these days. If you use code-server you would like it so much, saves time, you can code anywhere without carrying your host machine anywhere and also take benefits of cloud computes when required.

**Reply by lg-nineoneninethreenine** [reply]
So, how do I run it in Google Cloud?

**Reply by lg-nineoneninethreenine** [reply]
And, if I run it in Google Cloud, how can I access it?

**Reply by nooobcoder** [reply]
In Google cloud you need to deploy a VM and have to have a static public IP address to access it. Choose a debian server and get started. If you need more help with this, you can read docs about deploying VM and setting firewall rules on GCP (Google Cloud Platform), i use AWS so can't guide you much about Google cloud.

**Reply by lg-nineoneninethreenine** [reply]
I learnt how to do it, I will use Ubuntu method

**Reply by lg-nineoneninethreenine** [reply]
I can use WSL for this also?

**Reply by lg-nineoneninethreenine** [reply]
Like an alternative?

**Reply by nooobcoder** [reply]
WSL stands for Windows Subsystem for Linux, it runs an amateur version of Ubuntu under the hood. I have never used it, but afterall it is a linux VM, so you can use it to run code-server. 

@coder-lg Follow along this video to just get started with code-server (thanks to @bpmct )
https://www.youtube.com/watch?v=AATs7M4pZ9E

Let me tell you, this would not provide you a stable dev environment, until you get your head around with rclone and set up your needed dependencies at once in the docker file. **Everytime dockerfile is modified, a new environment is re-deployed**

But, hey you can try it once to get a taste of it. You can ofcourse contact back if you need to advance with the configuration and know more about deploying `code-server` with docker.

Thank You

**Reply by lg-nineoneninethreenine** [reply]
Thanks a lot! I am now able to access the code-server with Railway.app!
